dapp/src/services/apy.service.js

```diff
@@ -0,0 +1,35 @@
+import { apyDayOptions } from 'utils/constants'
+
+export default class ApyService {
+  async fetchApy() {
+    const dayResults = await Promise.all(
+      apyDayOptions.map(async (days) => {
+        let endpoint, varName
+        if (apyDayOptions.includes(days)) {
+          endpoint = `${process.env.APR_ANALYTICS_ENDPOINT}/${days}`
+          varName = `apy${days}`
+        } else {
+          throw new Error(`Unexpected days param: ${days}`)
+        }
+
+        try {
+          const response = await fetch(endpoint)
+          if (response.ok) {
+            const json = await response.json()
+            return json.apy / 100
+          }
+        } catch (err) {
```

Comment:
I would remove the whole try/catch statement. (and throw a new error when `!response.ok` as mentioned above.